aws-storage-s3/src/test/java/com/amplifyframework/storage/s3/StorageComponentTest.java

```diff
@@ -93,6 +97,48 @@ private static StorageCategoryConfiguration buildConfiguration() {
         return configuration;
     }
 
+    /**
+     * Test that calling get URL method from Storage category correctly invokes
+     * the registered AWSS3StoragePlugin instance and returns a {@link URL}
+     * instance for that download.
+     * @throws StorageException when an error is encountered while generating
+     *         URL from storage service
+     */
+    @Test
+    public void testGenerateUrlGetsPresignedUrl() throws StorageException {
+        final String fromRemoteKey = RandomString.string();
+        final URL urlFromRemoteKey;
+        try {
+            // URL instance cannot be mocked so just make one
+            // https://{random-host}:0/{fromRemoteKey}
+            urlFromRemoteKey = new URL(
```

Comment:
It would probably work without this random URL if you use Robolectric, which will provide an implementation of andorid's URL. (But, Robolectric is slow. So I do like your solution, too.)

You could pull this out to a `RandomURL.url()` test util! Like:
```
public static URL url() {
    try {
        return new URL(... params ...);
    } catch (MalformedURLException exception) {
        throw new RuntimeException(exception);
    }
}
```